move/seal_testnet/sources/key_server.move

```diff
@@ -0,0 +1,378 @@
+        id: object::new(ctx),
+        first_version: 2,
+        last_version: 2,
+    };
+
+    let key_server_v2 = KeyServerV2 {
+        id: object::new(ctx),
+        name,
+        key_type: KeyTypeBonehFranklinBLS12381,
+        pk,
+        server_type: ServerType::Committee { threshold },
+    };
+
+    df::add(&mut key_server.id, 2, key_server_v2);
+    key_server
+}
+
+/// Upgrade the current key server to v2, still a single owner object.
+public fun upgrade_to_v2(ks: &mut KeyServer, ctx: &mut TxContext) {
+    let key_server_v2 = KeyServerV2 {
+        id: object::new(ctx),
+        name: ks.v1().name,
+        key_type: ks.v1().key_type,
+        pk: ks.v1().pk,
+        server_type: ServerType::Independent { url: ks.v1().url },
+    };
+
+    df::add(&mut ks.id, 2, key_server_v2);
+    ks.last_version = 2;
+}
+
+/// Create and add partial key server objects for a committee-owned key server.
+public fun add_all_partial_key_servers(
+    key_server: &mut KeyServer,
+    members: &vector<address>,
+    partial_pks: &vector<vector<u8>>,
+    ctx: &mut TxContext,
+) {
+    assert_committee_server_v2(key_server);
+    let mut i = 0;
+    while (i < members.length()) {
+        let partial_pk = partial_pks[i];
+        let partial_key_server = PartialKeyServer {
+            id: object::new(ctx),
+            pk: partial_pk,
+            url: b"".to_string(), // intialize empty url, member can update this
+            party_id: (i as u16),
+            owner: members[i],
+        };
+
+        let v2: &mut KeyServerV2 = key_server.v2_mut();
+        df::add(&mut v2.id, members[i], partial_key_server);
+        i = i + 1;
+    };
+}
+
+/// Remove a single partial key server (for members leaving during rotation)
+public fun remove_partial_key_server(key_server: &mut KeyServer, member: address) {
+    assert_committee_server_v2(key_server);
+    let v2: &mut KeyServerV2 = key_server.v2_mut();
+
+    if (df::exists_(&v2.id, member)) {
+        let PartialKeyServer { id, pk: _, url: _, party_id: _, owner: _ } = df::remove<
+            address,
+            PartialKeyServer,
+        >(&mut v2.id, member);
+        object::delete(id);
+    };
+}
+
+/// Add a single partial key server (for new members during rotation)
+public fun add_partial_key_server(
+    key_server: &mut KeyServer,
+    member: address,
+    partial_pk: vector<u8>,
+    party_id: u16,
+    url: String,
+    ctx: &mut TxContext,
+) {
+    assert_committee_server_v2(key_server);
+    let partial_key_server = PartialKeyServer {
+        id: object::new(ctx),
+        pk: partial_pk,
+        url,
+        party_id,
+        owner: member,
+    };
+
+    let v2: &mut KeyServerV2 = key_server.v2_mut();
+    df::add(&mut v2.id, member, partial_key_server);
+}
+
+/// Update the URL of a partial key server, can only update the caller created server.
+public fun update_partial_ks_url(key_server: &mut KeyServer, url: String, ctx: &mut TxContext) {
```

Comment:
let's change that to a given account (and the code in committee would set it to ctx.sender) since other use cases might not want to call it for ctx.sender()